examples/str/consumer/Pulumi.yaml

```diff
@@ -1,30 +1,36 @@
 name: test-str
 runtime: yaml
+
+plugins:
+  providers:
+    - name: str
+      path: ..
+
 variables:
   replaced:
     Fn::Invoke:
-      Function: str:Replace
+      Function: str:replace
```

Comment:
Is this because the str provider or YAML supports both casings, or because this provider is using a lowercase token now and requires it?

Asking as this appears inconsistent with tokens used elsewhere which - for whatever reason - capitalize the type at the end.

Reply:
The `str` provider only supports a single casing, which is lowercase for functions. Pulumi YAML doesn't change anything there.

I changed functions to expose a lowercase name (`pkg:module:name`) since that is consistent with function type tokens in other packages. Function tokens follow a different convention than other resource tokens. I'm not sure why. This change was made a while ago, and the example was broken. Since it wasn't tested, I didn't notice. This PR adds tests to the examples.